src/refract/VisitorUtils.h

```diff
@@ -125,32 +125,68 @@ namespace refract
         }
     };
 
-    struct GetEnumValue {
+    template<>
+    struct GetValue<refract::EnumElement, typename refract::EnumElement::ValueType> {
+        using EnumElement = refract::EnumElement;
         const EnumElement& element;
 
-        GetEnumValue(const EnumElement& e) : element(e) {}
+        GetValue(const EnumElement& e) : element(e) {}
 
-        operator const EnumElement::ValueType*() {
+        operator const IElement*() {
+            return GetEnumValue(element);
+        }
+
+        IElement* GetEnumValue(const EnumElement& element) {
             if (const EnumElement* s = GetSample(element)) {
-                return &s->value;
+                return GetEnumValue(*s);
```

Comment:
I don't understand why this would be changed to an enum specific extraction of the sample value. Now that enum default/sample/value behaviour is identical to other elements surely it could have the same behaviour, except for falling back to `enumerations` when no values/samples/defaults are found?

Reply:
There is difference between Enum and Other Elements. For Enum you need a pick one value from possible values.

Another "complication" is we want to be backward compatible (at least first time of iteration) before redefine Enum behavior

Reply:
> For Enum you need a pick one value from possible values.

That should not be true in the new form, `default` is an element and `samples` is an array of samples (array of elements) so this matches other element behaviour:

https://github.com/apiaryio/api-elements/pull/28

Reply:
Then we will have "non compatible" building of elements default/samples for enum. All others contain full element in default/sample.

And it mean there will be again incompatibility for calling `GetSample()`, `GetDefault()` voth returns Same Type of Element

Reply:
You typically call GetSample/GetValue and pick value from them.
```
ArrayElement* a = GetDefault<ArrayElement>(element);
if (a->empty()) {
  return a->value;
}
```